pkg/registry/bundlegraphloader_test.go

```diff
@@ -312,3 +313,71 @@ func TestBundleGraphLoader(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSkipPatchCandidate(t *testing.T) {
+	tests := []struct{
+		name     string
+		added     string
+		compare  string
+		expected bool
+	}{
+		{
+			name: "equal versions",
+			added: "0.0.0",
+			compare: "0.0.0",
+			expected: false,
+		},
+		{
+			name: "do not accept different major/minor version",
+			added: "0.0.0",
+			compare: "0.0.0",
+			expected: false,
+		},
+		{
+			name: "do not accept smaller patch version",
+			added: "0.0.0",
+			compare: "0.0.1",
+			expected: false,
+		},
+		{
+			name: "accept larger patch version",
+			added: "0.0.1",
+			compare: "0.0.0",
+			expected: true,
+		},
+		{
+			name: "accept patch version without pre-release",
+			added: "0.0.0",
+			compare: "0.0.0-1",
+			expected: true,
+		},
+		{
+			name: "accept longer pre-release with same prefix",
+			added: "0.0.1-1.2",
+			compare: "0.0.1-1",
+			expected: true,
+		},
+		{
+			name: "accept lexicographically larger pre-release",
+			added: "0.0.1-11",
+			compare: "0.0.1-2",
+			expected: true,
+		},
+		{
+			name: "accept lexicographically larger pre-release",
```

Comment:
NIt: It might be nice to give this subtest a distinct name to play nicely with `-run` and have clearer test run outputs.